v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -118,10 +117,14 @@ func (ctx KubeGlobalContext) ForTest(t *testing.T) *KubePerTestContext {
 	// Test configs never want SyncPeriod set as it introduces jitter
 	cfg.SyncPeriod = nil
 
-	return ctx.ForTestWithConfig(t, cfg)
+	return ctx.forTestWithConfig(t, cfg, false)
 }
 
 func (ctx KubeGlobalContext) ForTestWithConfig(t *testing.T, cfg config.Values) *KubePerTestContext {
+	return ctx.forTestWithConfig(t, cfg, true)
+}
+
+func (ctx KubeGlobalContext) forTestWithConfig(t *testing.T, cfg config.Values, countsTowardsLimit bool) *KubePerTestContext {
```

Comment:
See above comment. The value `countsTowardsLimit` is intelligible here in the implementation, but at callsites it's mysterious and cryptic - which is cool in Film Noir, but less so in code.